### ISSUE TITLE
Two Way binding for decimals causes System.AccessViolationException on property changed

### ISSUE THREAD

**Issue by eric-vogt-pacesetter** [microsoft/microsoft-ui-xaml#9064]
### Describe the bug

If you add x:Bind for a decimal property when you update the value of the decimal it causes System.AccessViolationException.

### Steps to reproduce the bug

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/131187967/e434c604-5f4f-45a7-af75-3e0f475c6bc7)
![image](https://github.com/microsoft/microsoft-ui-xaml/assets/131187967/65dae7d7-33fd-4385-85b6-40111d2131bf)
Try to update the value of the number box in the application
![image](https://github.com/microsoft/microsoft-ui-xaml/assets/131187967/b36e6bfa-edb5-4ccd-bd30-79b9d15868d3)

### Expected behavior

Should not cause exception

### Screenshots

See above

### NuGet package version

WinUI 3 - Windows App SDK 1.4.2: 1.4.231008000

### Additional context

Windows 10 version 22H2 (19045, 2022 Update)
Visual Studio 2022

**Comment by kmgallahan** [comment]
> Try to update the value of the number box in the application

1. Your screenshot indicates you are trying to bind to `NumberBox.Text`, not `NumberBox.Value`
2. [`NumberBox.Value`](https://learn.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.numberbox.value) is a `double`, not a `decimal`

**Comment by kmgallahan** [comment]
You are trying to bind a `decimal` to a `string`, for which there is no build-in TwoWay converter.

On the initial get a string can be retrieved because:

> If you bind a text control to a value that is not a string, the data binding engine will convert the value to a string. If the value is a reference type, the data binding engine will retrieve the string value by calling [ICustomPropertyProvider.GetStringRepresentation](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.data.icustompropertyprovider.getstringrepresentation) or [IStringable.ToString](https://learn.microsoft.com/en-us/windows/desktop/api/windows.foundation/nf-windows-foundation-istringable-tostring) if available, and will otherwise call [Object.ToString](https://learn.microsoft.com/en-us/dotnet/api/system.object.tostring#System_Object_ToString).

However, on set it will crash with the error `The value cannot be converted to type Decimal.` This happens in native code and can be observed if you use mixed mode debugging.

**Comment by eric-vogt-pacesetter** [comment]
Thank you very much for the explanation! Value is a double type which can't be bound to a decimal so it seems like there is currently no solution to binding a decimal to a number box
![image](https://github.com/microsoft/microsoft-ui-xaml/assets/131187967/e06d503e-edf9-4591-8b43-6ab8cb7197c2)

**Comment by kmgallahan** [comment]
That is correct, you must use a converter if you want to work with a decimal instead of the double in code behind / elsewhere.

**Comment by Scottj1s** [comment]
@eric-vogt-pacesetter Is the solution suggested by kmgallahan acceptable?
